src/network/netdev/macvlan.c

```diff
@@ -52,6 +54,13 @@ static int netdev_macvlan_fill_message_create(NetDev *netdev, Link *link, sd_net
                         return log_netdev_error_errno(netdev, r, "Could not append IFLA_MACVLAN_MODE attribute: %m");
         }
 
+        // set the nopromisc flag if Promiscuous= of the link is explicitly set to false
+        if (m->mode == NETDEV_MACVLAN_MODE_PASSTHRU && !link->network->promiscuous) {
```

Comment:
`link->network->promiscuous` is a tristate flag, and do not use implicit cast to boolean. So, use the following
```
if (m->mode == NETDEV_MACVLAN_MODE_PASSTHRU && link->network->promiscuous == 0) {
```
Or, maybe,
```
if (m->mode == NETDEV_MACVLAN_MODE_PASSTHRU && link->network->promiscuous >= 0) {
        r = sd_netlink_message_append_u16(req, IFLA_MACVLAN_FLAGS, link->network->promiscuous == 0 ? MACVLAN_FLAG_NOPROMISC : 0);
```
i.e. also enable promiscuous mode explicitly when `Promiscuous=yes`?

Reply:
I don't think we should / need to change the flags of the macvlan. AFAIK, it only prevents the instruction from the macvlan side anyway.